oasislmf/execution/runner.py

```diff
@@ -109,6 +110,43 @@ def custom_get_getmodel_cmd(
     logging.info(bash_trace.decode('utf-8'))
 
 
+def rerun():
+    """
+    A function to find where an error was made and to rerun that part of the script without
+    NumBa to give better error messages
+    """
+    try:
+        with open("event_error.json", "r") as f:
+            event_error = json.load(f).get("event_id")
+    except FileNotFoundError:
+        return
+
+    env = os.environ.copy()
+    env['NUMBA_DISABLE_JIT'] = "1"
+    eve_cmd = f"printf 'event_id\n {event_error}\n' | evetobin"
+    ktools_pipeline = ''
+
+    with open("run_ktools.sh", "r") as bash_script:
+        for line in bash_script:
+            if "( ( eve" in line:
+                ktools_pipeline = line.split('|')
+                break
+
+    gul_cmd = [cmd.strip() for cmd in ktools_pipeline if cmd.strip().startswith(('gul'))].pop(0)
+    fm_cmd = [cmd.strip() for cmd in ktools_pipeline if cmd.strip().startswith(('fm'))].pop(0)
+    pipe_output = "/tmp/il_P1"
+    summary_output = "/tmp/il_S1_summary_P1"
+    single_event_pipe = f"{eve_cmd} | {gul_cmd} | {fm_cmd} -o {pipe_output}"
+    summary_pipe = f"summarypy -t il -m -1 {summary_output} < {pipe_output}"
+    try:
+        with open("rerun_errors.log", "w") as error_log:
```

Comment:
in order to have intermediary output, to ease bug reproduction, can you save each intermediary output.
{eve_cmd} | {gul_cmd} > {event_id}_gul.bin
then if fmpy is present 
 {fm_cmd} -o {pipe_output} -i {event_id}_gul.bin 
 and same for the reinsurance fmpy